velox/functions/sparksql/Hash.cpp

```diff
@@ -54,36 +295,16 @@ void applyWithType(
           decoded->nulls(&rows), rows.begin(), rows.end());
       selected = selectedMinusNulls.get();
     }
-    switch (args[i]->type()->kind()) {
-// Derived from InterpretedHashFunction.hash:
-// https://github.com/apache/spark/blob/382b66e/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/hash.scala#L532
-#define CASE(typeEnum, hashFn, inputType)                                      \
-  case TypeKind::typeEnum:                                                     \
-    selected->applyToSelected([&](int row) {                                   \
-      result.set(                                                              \
-          row, hashFn(decoded->valueAt<inputType>(row), result.valueAt(row))); \
-    });                                                                        \
-    break;
-      CASE(BOOLEAN, hash.hashInt32, bool);
-      CASE(TINYINT, hash.hashInt32, int8_t);
-      CASE(SMALLINT, hash.hashInt32, int16_t);
-      CASE(INTEGER, hash.hashInt32, int32_t);
-      CASE(BIGINT, hash.hashInt64, int64_t);
-      CASE(VARCHAR, hash.hashBytes, StringView);
-      CASE(VARBINARY, hash.hashBytes, StringView);
-      CASE(REAL, hash.hashFloat, float);
-      CASE(DOUBLE, hash.hashDouble, double);
-      CASE(HUGEINT, hash.hashLongDecimal, int128_t);
-      CASE(TIMESTAMP, hash.hashTimestamp, Timestamp);
-#undef CASE
-      default:
-        VELOX_NYI(
-            "Unsupported type for HASH(): {}", args[i]->type()->toString());
-    }
+
+    auto hasher = createVectorHasher<HashClass>(*decoded);
+    selected->applyToSelected([&](auto row) {
+      result.set(row, hasher->hashNotNullAt(row, result.valueAt(row)));
+    });
```

Comment:
@marin-ma Looks we only need to keep this switch case statement for primitive type. Let's keep this to address @mbasmanova 's perf concern.

Let's test TPCH firstly.